Microsoft.ILVerification depends on too low version of System.Reflection.Metadata package

### Description

The Microsoft.ILVerification/6.0.0 package depends on System.Reflection.Metadata (>= 1.8.1).  lib/netstandard2.0/ILVerification.dll references "System.Reflection.Metadata, version=5.0.0.0, publicKeyToken=b03f5f7f11d50a3a".  However, in the System.Reflection.Metadata/1.8.1 package, all lib\\*\System.Reflection.Metadata.dll files are versioned as "System.Reflection.Metadata, Version=1.4.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". This assembly version 1.4.5.0 is lower than the referenced version 5.0.0.0.

The System.Reflection.Metadata/5.0.0 package contains assemblies versioned as "System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". This would satisfy the assembly reference, but Microsoft.ILVerification/6.0.0 does not depend on this version of the package.

### Reproduction Steps

Ilve.csproj:

```XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net48</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.ILVerification" Version="6.0.0" />
  </ItemGroup>

</Project>
```

Class1.cs:

```C#
﻿public class Class1
{
    void M()
    {
        var verifier = new ILVerify.Verifier(null);
        verifier.Verify(null);
    }
}
```

dotnet build

### Expected behavior

The build system chooses a compatible version of the System.Reflection.Metadata package, and the build succeeds. (An ArgumentNullException might be thrown if you try to run the result.)

### Actual behavior

The build fails.

```
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277: Found conflicts between different versions of "System.Reflection.Metadata" that could not be resolved. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277: There was a conflict between "System.Reflection.Metadata, Version=1.4.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     "System.Reflection.Metadata, Version=1.4.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     References which depend on "System.Reflection.Metadata, Version=1.4.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll]. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:         C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:           Project file item includes which caused reference "C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:             C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     References which depend on "System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" []. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:         C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:           Project file item includes which caused reference "C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:             C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277: Found conflicts between different versions of "System.Collections.Immutable" that could not be resolved. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277: There was a conflict between "System.Collections.Immutable, Version=1.2.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Collections.Immutable, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     "System.Collections.Immutable, Version=1.2.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "System.Collections.Immutable, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     References which depend on "System.Collections.Immutable, Version=1.2.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Users\Kalle\.nuget\packages\system.collections.immutable\1.7.1\lib\net461\System.Collections.Immutable.dll]. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:         C:\Users\Kalle\.nuget\packages\system.collections.immutable\1.7.1\lib\net461\System.Collections.Immutable.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:           Project file item includes which caused reference "C:\Users\Kalle\.nuget\packages\system.collections.immutable\1.7.1\lib\net461\System.Collections.Immutable.dll". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:             C:\Users\Kalle\.nuget\packages\system.collections.immutable\1.7.1\lib\net461\System.Collections.Immutable.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:     References which depend on "System.Collections.Immutable, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" []. [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:         C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:           Project file item includes which caused reference "C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:             C:\Users\Kalle\.nuget\packages\microsoft.ilverification\6.0.0\lib\netstandard2.0\ILVerification.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:         C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:           Project file item includes which caused reference "C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll". [C:\Projects\Kalle\Ilve\Ilve.csproj]
C:\Program Files\dotnet\sdk\6.0.101\Microsoft.Common.CurrentVersion.targets(2304,5): warning MSB3277:             C:\Users\Kalle\.nuget\packages\system.reflection.metadata\1.8.1\lib\net461\System.Reflection.Metadata.dll [C:\Projects\Kalle\Ilve\Ilve.csproj]
CSC : error CS1705: Assembly 'ILVerification' with identity 'ILVerification, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Reflection.Metadata' with identity 'System.Reflection.Metadata, Version=1.4.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' [C:\Projects\Kalle\Ilve\Ilve.csproj]

Build FAILED.
```

### Regression?

Don't know.

### Known Workarounds

```XML
<PackageReference Include="System.Reflection.Metadata" Version="5.0.0" />
```

### Configuration

.NET SDK 6.0.101 targeting .NET Framework 4.8 on Windows 10.0.19044.1466 (x64).

### Other information

ILVerification.csproj is built against System.Reflection.Metadata/5.0.0:

<https://github.com/dotnet/runtime/blob/3a25a7f1cc446b60678ed25c9d829420d6321eba/src/coreclr/tools/ILVerification/ILVerification.csproj#L15-L17>

<https://github.com/dotnet/runtime/blob/3a25a7f1cc446b60678ed25c9d829420d6321eba/eng/Versions.props#L99>

However, Microsoft.ILVerification.pkgproj references System.Reflection.Metadata/1.8.1 instead:

<https://github.com/dotnet/runtime/blob/3a25a7f1cc446b60678ed25c9d829420d6321eba/src/coreclr/.nuget/Microsoft.ILVerification/Microsoft.ILVerification.pkgproj#L17-L21>

--- a/src/coreclr/.nuget/Microsoft.ILVerification/Microsoft.ILVerification.pkgproj
+++ b/src/coreclr/.nuget/Microsoft.ILVerification/Microsoft.ILVerification.pkgproj
@@ -15,7 +15,7 @@
        <TargetPath>lib\netstandard2.0\ILVerification.dll</TargetPath>
     </PackageFile>
     <Dependency Include="System.Reflection.Metadata">
-       <Version>1.8.1</Version>
+       <Version>$(SystemReflectionMetadataVersion)</Version>
        <TargetFramework>netstandard2.0</TargetFramework>
        <Exclude>Build,Analyzers</Exclude>
     </Dependency>

